juju/main.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import yaml
+import importlib
+
+CHARM_STATE_FILE = '.unit-state.db'
+
+
+def debugf(format, *args, **kwargs):
+    pass
+
+
+class InvalidHookEntry(Exception):
+    pass
+
+
+def _get_charm_dir():
+    charm_dir = os.environ.get("JUJU_CHARM_DIR")
+    if charm_dir is None:
+        # Assume $JUJU_CHARM_DIR/lib/juju/main.py structure.
+        charm_dir = os.path.dirname(os.path.dirname(
+            os.path.dirname(os.path.abspath(__file__))))
+    return charm_dir
+
+
+def _load_metadata():
```

Comment:
Could move those to `metadata.py` or something else later so that other code could use it (including developer-provided code).